helper/funcs.go

```diff
@@ -503,3 +503,12 @@ func Merge[T comparable](a, b T) T {
 	}
 	return a
 }
+
+// AddClamped adds integers but returns maxint if there's overflow
+func AddClamped(a, b int64) int64 {
+	c := a + b
+	if (c > a) == (b > 0) {
+		return c
+	}
+	return math.MaxInt64
+}
```

Comment:
The best time to plant a tree was 20 years ago, the next best time is today.

If anything I might put this in a separate package so you'd have something like
`safemath.Add(a, b)` instead of `helper.AddClamped(a, b)`

Reply:
Done!